### PR TITLE
feat: add more context to invalid text string parents

### PR DESCRIPTION
**Pull request by EvanBacon**
## Summary

The stack traces shown in LogBox often just point to fast refresh and don't provide much context into cases where a user adds a string outside of a Text element, this PR improves discoverability a little by adding the invalid text to the error message.

## Changelog

[General] [Added] - Improved invalid text parent error message.

## Test Plan

Tried it in a basic case: `<View>foobar</View>` will throw: `Error: Text strings "foobar" must be rendered within a <Text> component.`

### CODE REVIEW

**Comment by ecreeth** [comment]
I believe that these changes must be applied in the react repo [ReactNativeHostConfig](https://github.com/facebook/react/blob/main/packages/react-native-renderer/src/ReactNativeHostConfig.js#L148) & [ReactFabricHostConfig](https://github.com/facebook/react/blob/main/packages/react-native-renderer/src/ReactFabricHostConfig.js#L265)

**Comment by yungsters** [comment]
What @ecreeth said, but also… check out these two PRs by me:

- https://github.com/facebook/react/pull/19581
- https://github.com/facebook/react/pull/19723

tl;dr — I previously attempted a similar change, but was recommended to revert it because @acdlite and @sebmarkbage raised concerns with logging arbitrary strings which may be sensitive (e.g. passwords, banking information).

**Comment by Yonom** [comment]
Perhaps a fair compromise is to log this information in debug builds only?

**Comment by yungsters** [comment]
@Yonom — Some people unknowingly run debug builds in production (e.g. when you see RedBox displaying in some production apps), so there is still some risk there.

@EvanBacon — What do you think about only appending the string if an opt-in flag is set (e.g. `global.debugInvalidTextStrings`), and then changing the error message to inform people to do that to opt into the extra error text? (And maybe when the opt-in flag is set, remind them to remove it when they're done debugging.)

**Comment by yungsters** [comment]
I am going to close this PR because the change has to be made in the React repository, as @ecreeth suggested above.

**Comment by EvanBacon** [comment]
Opened a PR for adding to react proper.

> tl;dr — I previously attempted a similar change, but was recommended to revert it because @acdlite and @sebmarkbage raised concerns with logging arbitrary strings which may be sensitive (e.g. passwords, banking information).

I don't quite understand this issue since the error would be thrown when attempting to explicitly render a value to the screen, e.g. someone wrapped their password in JSX and passed it to the render method lol.

**Comment by yungsters** [comment]
Yeah, that is the concern.

Either by design or by mistake, someone could conceivably render sensitive information (e.g. password, social security number, credit card information) as text without `<Text>`. Since errors are typically persisted to storage in plaintext, logging these strings is more likely to lead to unknowingly storing sensitive information in unsecured storage solutions.

**Comment by yungsters** [comment]
For posterity, https://github.com/facebook/react/pull/22725 seems to be the React PR. Let's move future discussion about this contribution to that repository.
